common/lib/client/realtimechannel.js

```diff
@@ -441,6 +441,13 @@ var RealtimeChannel = (function() {
 			break;
 
 		case actions.MESSAGE:
+
+			//RTL17
+			if(this.state !== 'attached') {
+				Logger.logAction(Logger.LOG_MAJOR, 'RealtimeChannel.onMessage()', 'Message skipped. Channel not in ATTACHED state.');
```

Comment:
This message could be more helpful for debugging - say,
`Logger.logAction(Logger.LOG_MAJOR, 'RealtimeChannel.onMessage()', 'Message ' + message.id + ' skipped as channel state is ' + channel.state;`

Reply:
https://github.com/ably/ably-js/pull/638/commits/f76fc914dc88b40a0575734a4f3fed5137416684